### PR TITLE
feat: save clusterfile to the cluster 

### PR DESCRIPTION
**Pull request by kakaZhou719**
### Describe what this PR does / why we need it

at present , we store the cluster configuration information on the master0 local disk after we create the cluster, such as node auth info , cluster plugins, cluster env and so on .

If this file is mistakenly deleted or modified by mistake, it will be a great disaster for the maintenance of the cluster. 

so we need to store this file in the cluster to do disaster preparedness, and each time we can prioritized use the clusterfile  in the cluster for operation.

### Describe how you did it

1. save clusterfile to cluster as configmap with name "sealer-clusterfile"  in namespace "kub-system".
2. modify some cmd examples

### CODE REVIEW

File: pkg/clusterfile/clusterfile.go [hunk 1]
```diff
@@ -150,27 +176,72 @@ func (c *ClusterFile) SaveAll() error {
...
+	configMap.Data = map[string]string{configMapDataName: string(data)}
+
+	_, err = client.ConfigMap(configMapNamespace).Apply(context.TODO(), configMap, metav1.ApplyOptions{FieldManager: "kubectl-create"})
+	if err != nil {
+		return fmt.Errorf("failed to create configmap: %v", err)
 	}
+	return nil
 }
 
 func NewClusterFile(b []byte) (Interface, error) {
```
Comment (VinceCui) [inline pkg/clusterfile/clusterfile.go:208]: good design~

File: pkg/clusterfile/clusterfile.go [hunk 2]
```diff
@@ -74,7 +89,18 @@ func (c *ClusterFile) CommitSnapshot() {
 	*c.cfSnapshot = *c.cluster
```
Comment (VinceCui) [inline pkg/clusterfile/clusterfile.go:89]: Although this is not part of this PR, i want to say the cfSnapshot maybe a bad design.

Why we need a snapshot of a cluster file? I think the only trusty cluster status is not the clusterfile but the k8s, so i think we no need control the version of clusterfile carefully.

File: pkg/clusterfile/clusterfile.go [hunk 3]
```diff
@@ -30,17 +34,28 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
+const (
+	configMapNamespace = "kube-system"
+	configMapDataName  = "Clusterfile"
+	configMapName      = "sealer-clusterfile"
+)
+
 type Interface interface {
```
Comment (VinceCui) [inline pkg/clusterfile/clusterfile.go:43]: Shall we save the application.json into k8s too?

File: cmd/sealer/cmd/cluster/delete.go [hunk 4]
```diff
@@ -96,18 +93,28 @@ func NewDeleteCmd() *cobra.Command {
 }
 
 func deleteCluster(workClusterfile string) error {
```
Comment (VinceCui) [inline cmd/sealer/cmd/cluster/delete.go:95]: good design.

File: cmd/sealer/cmd/cluster/scale-up.go [hunk 5]
```diff
@@ -39,8 +38,8 @@ User can scale-up cluster by explicitly specifying host IP`
 
 var exampleForScaleUpCmd = `
 scale-up cluster:
-  sealer join --masters 192.168.0.1 --nodes 192.168.0.2 -p Sealer123
-  sealer join --masters 192.168.0.1-192.168.0.3 --nodes 192.168.0.4-192.168.0.6 -p Sealer123
+  sealer scale-up --masters 192.168.0.1 --nodes 192.168.0.2 -p Sealer123
```
Comment (VinceCui) [inline cmd/sealer/cmd/cluster/scale-up.go:41]: Are you find that, when you change the join.go or apply.go, you need change this file too. This is a bad design, you should use a common function for both command join, scale-up and apply. 

When should we refactor the code? The answer is NOW.

File: cmd/sealer/cmd/cluster/apply.go [hunk 6]
```diff
@@ -81,7 +81,7 @@ func NewApplyCmd() *cobra.Command {
 			}
 
 			//save desired clusterfile
-			if err = cf.SaveAll(); err != nil {
```
[the remark this answers is not in the document]
Comment (VinceCui) [reply cmd/sealer/cmd/cluster/apply.go:84]: If this clusterfile is an App File, should we save it?

**Comment by VinceCui** [comment]
Maybe we can provide SDK for read the ClusterFile from k8s.

File: pkg/clusterfile/clusterfile.go (hunk 3 above)
Comment (kakaZhou719) [reply pkg/clusterfile/clusterfile.go:43]: i will do it in the another PR.

File: cmd/sealer/cmd/cluster/apply.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (kakaZhou719) [reply cmd/sealer/cmd/cluster/apply.go:84]: Mainly to store the expected cluster node information temporarily.  if apply failed, will use it to delete the cluster node ,if not, new save all will overwrite it.

[the remark this answers is not in the document]
Comment (kakaZhou719) [reply cmd/sealer/cmd/cluster/apply.go:84]: > If this clusterfile is an App File, should we save it?

yes, if it is app, no need to pre save expected clusterfile.

**Review by VinceCui — APPROVED** [review]
